src/llm_vm/onsite_llm.py

```diff
@@ -527,3 +485,56 @@ def finetune(self, dataset, optimizer, c_id, small_model_filename=None):
         # optimizer.storage.set_training_in_progress(c_id, False)
         # if old_model is not None:
         #     openai.Model.delete(old_model)
+
+
+class BaseCtransformers(BaseOnsiteLLM):
+    """
```

Comment:
More generally, do we need to do anything with the abstract base class stuff here?

Reply:
i'm moreso wondering if we should specify the model file property in this "base" class, it need not be a new base class, since its properly a subtype of our general base class

Reply:
i've added a bit more generalization so now all logic is in the base class. only model_uri is changing for instances

Reply:
> i'm moreso wondering if we should specify the model file property in this "base" class, it need not be a new base class, since its properly a subtype of our general base class

ow, ok. my bad, i completely misunderstood this

Reply:
nah, i was being vague :)